src/test/java/com/microsoft/sqlserver/jdbc/bvt/bvtTest.java

```diff
@@ -1,678 +1,587 @@
-
-		// guarantees selectMethod=cursor
-		cursor = true;
-		try {
-			stmt = conn().createStatement();
-
-			// SELECT * FROM <table2> ORDER BY <key>
-			String pk = Tables.primaryKey();
-			String query = Tables.select_Orderby(table2, pk);
-			rs = stmt.executeQuery(query);
-			bvt_rs = new bvt_ResultSet(rs);
-
-			// verify resultSet
-			bvt_rs.verify();
-			cursor = false;
-		} finally {
-			terminateVariation();
-		}
-	}
-
-	@AfterClass
-	public static void terminate() throws SQLException {
-
-		try {
-			stmt = conn().createStatement();
-			stmt.executeUpdate(Tables.dropTable(table2));
-			stmt.executeUpdate(Tables.dropTable(table1));
-		} finally {
-			terminateVariation();
-		}
-	}
-
-	public static String getConnectionURL() {
-		connectionUrl = System.getenv("mssql_jdbc_test_connection_properties");
-		
-		if(null == connectionUrl){
-			fail("Please setup environment variable mssql_jdbc_test_connection_properties");
-		}
-		
-		if (cursor){
-			connectionUrl += ";selectMethod=cursor;";
-		}
-		
-		if (querytimeout){
-			connectionUrl += ";queryTimeout=10";
-		}
-
-		return connectionUrl;
-	}
-
-	public static Connection conn() {
-
-		connectionUrl = getConnectionURL();
-		// Establish the connection.
-		try {
-
-			Connection con = DriverManager.getConnection(connectionUrl);
-			return con;
-			// Handle any errors that may have occurred.
-		} catch (SQLException e) {
-			fail("Please make sure the environment variable mssql_jdbc_test_connection_properties is set with correct connection properties.\n" 
-					+ e.toString());
-		}
-		return null;
-	}
-
-	public static void terminateVariation() throws SQLException {
-		if (con != null && !con.isClosed()) {
-			try {
-				con.close();
-			} catch (SQLException e) {
-				fail("Connection close threw : " + e.toString());
-			} finally {
-				if (null != bvt_rs)
-					bvt_rs.close();
-				if (null != rs)
-					rs.close();
-				if (null != stmt)
-					stmt.close();
-				if (null != pstmt)
-					pstmt.close();
-			}
-		}
-	}
+    @Test
+    public void testCreateStatementWithQueryTimeout() throws SQLException {
+
+        try {
+            conn = new DBConnection(connectionString + ";querytimeout=10");
+            stmt = conn.createStatement();
+            assertEquals(10, stmt.getQueryTimeout());
+        }
+        finally {
+            terminateVariation();
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // Create a statement
+    // ResultSet.Type_forward_only,
+    // ResultSet.CONCUR_READ_ONLY, executeQuery
+    // verify cursor by using next and previous and verify data
+    ///////////////////////////////////////////////////////////////////
+    @Test
+    public void testStmtForwardOnlyReadOnly() throws SQLException, ClassNotFoundException {
+
+        try {
+            conn = new DBConnection(connectionString);
+            stmt = conn.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
```

Comment:
Do you think it will be better to have an enum in DBStatement/DBConnection that has all combinations of resultSetType and resultSetConcurrency? So we don't have to call and set each type (will be useful to reuse in other tests), and just have validation in btvTest